src/components/MakeWidget/subcomponents/RateField/RateField.tsx

```diff
@@ -0,0 +1,54 @@
+import { useState } from "react";
+
+import BigNumber from "bignumber.js";
+
+import IconButton from "../../../IconButton/IconButton";
+import { Text, Wrapper, RateBox } from "./RateField.styles";
+
+function displayRate(rate: number) {
+  if (rate > 1000000) {
+    return ">1M";
+  }
+
+  if (rate < 0.05) {
+    return "<0.1";
+  }
+
+  return rate.toFixed(1);
+}
+
+export type RateFieldProps = {
+  token1: string;
+  token2: string;
+  rate: BigNumber;
+  className: string;
+};
+
+export const RateField: React.FC<RateFieldProps> = ({
+  token1,
+  token2,
+  rate,
+  className,
+}) => {
+  const [TokenPair, setTokenPair] = useState([token1, token2]);
+  const [CurrentRate, setCurrentRate] = useState(rate.toNumber());
```

Comment:
```suggestion
  const [currentRate, setCurrentRate] = useState(rate);
```

By converting BigNumber to number here you still have the [IEEE 754 double-precision binary floating point](https://en.wikipedia.org/wiki/Double-precision_floating-point_format) problem.

Read more about it here: https://docs.ethers.io/v5/api/utils/bignumber/